Assets/MixedRealityToolkit/_Core/Utilities/Scripts/Singleton.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Utilities
+{
+    /// <summary>
+    /// Singleton behavior class, used for components that should only have one instance.
+    /// <remarks>Singleton classes live on through scene transitions and will mark their 
+    /// parent root GameObject with <see cref="Object.DontDestroyOnLoad"/></remarks>
+    /// </summary>
+    /// <typeparam name="T">The Singleton Type</typeparam>
+    [DisallowMultipleComponent]
+    public abstract class Singleton<T> : MonoBehaviour where T : Singleton<T>
```

Comment:
I'm a little apprehensive to putting this class into vNEXT.  I thought we were going to move away from the Singleton pattern.

Reply:
I removed this class from the project.  Most of its guts now live in MixedRealityManager. It's the only class that will be a singleton tied to a game object reference in the scene.  All MRTK managers will be pure c# classes managed by this instance.

Reply:
I may disagree on this.  While I agree and accept that in the MRTK-vNext, the only class using the Singleton will be the manager.
However, that's not to say that other projects will already be consuming the singleton from the MRTK for their own projects, so it's still "useful" (even though we don't like it) for consumers projects.   Thoughts?

Reply:
Is rather see ya move away from it because it's caused us too much pain in the past, and the manager system is supposed to be the way forward. It'll reduce confusion and promote adoption of the system when people try to add to it.